pkg/cfn/manager/create_tasks.go

```diff
@@ -22,6 +22,16 @@ func (c *StackCollection) NewTasksToCreateClusterWithNodeGroups(nodeGroups []*ap
 			supportsManagedNodes: supportsManagedNodes,
 		},
 	)
+	var postClusterTasks *TaskTree
+	if len(postClusterCreationTasks) > 0 {
+		postClusterTasks = &TaskTree{Parallel: true}
```

Comment:
I'm thinking that since the tasks come from the caller, perhaps it should be the caller's responsibility to know whether they  can run in parallel or not. WDYT?

Reply:
hmm actually perhaps it's a reasonable assumption that the `postClusterCreationTasks` can run in parallel with the node tasks?

Reply:
hopefully it's a bit easier to see what's going on now